shared/src/managed-lib/ManagedLoader/Datadog.AutoInstrumentation.ManagedLoader/public/AssemblyLoader.cs

```diff
@@ -412,104 +412,62 @@ private static void GetTracerManagedBinariesDirectories(List<string> binaryDirs)
-
-            string nativeProfilerLoaderAssemblyFile;
-            if (isCoreFx)
-            {
-                nativeProfilerLoaderAssemblyFile = ReadEnvironmentVariable(Environment.Is64BitProcess ? "CORECLR_PROFILER_PATH_64" : "CORECLR_PROFILER_PATH_32");
-                nativeProfilerLoaderAssemblyFile = nativeProfilerLoaderAssemblyFile ?? ReadEnvironmentVariable("CORECLR_PROFILER_PATH");
-            }
-            else
-            {
-                nativeProfilerLoaderAssemblyFile = ReadEnvironmentVariable(Environment.Is64BitProcess ? "COR_PROFILER_PATH_64" : "COR_PROFILER_PATH_32");
-                nativeProfilerLoaderAssemblyFile = nativeProfilerLoaderAssemblyFile ?? ReadEnvironmentVariable("COR_PROFILER_PATH");
-            }
+            string profilerHomeDirectory = ReadEnvironmentVariable("DD_DOTNET_PROFILER_HOME");
 
             // Be defensive against env var not being set.
-            if (String.IsNullOrWhiteSpace(nativeProfilerLoaderAssemblyFile))
+            if (string.IsNullOrWhiteSpace(profilerHomeDirectory))
             {
                 return;
             }
 
-            string nativeProfilerLoaderAssemblyDirectory = Path.GetDirectoryName(nativeProfilerLoaderAssemblyFile);
+            string managedBinariesSubdir = GetRuntimeBasedProductBinariesSubdir();
+            string managedBinariesDirectory = Path.Combine(profilerHomeDirectory, managedBinariesSubdir);
 
+            if (binaryDirs != null && !string.IsNullOrWhiteSpace(managedBinariesDirectory))
             {
-                // OPTION A from above (SxS with Tracer):
-
-                string tracerHomeDirectory = nativeProfilerLoaderAssemblyDirectory;                             // Shared Tracer/Profiler loader is in Tracer HOME
-                string profilerHomeDirectory = Path.Combine(tracerHomeDirectory, "ContinuousProfiler");         // Profiler-HOME is in <Tracer-HOME>/ContinuousProfiler
-
-                string managedBinariesDirectory = Path.Combine(profilerHomeDirectory, managedBinariesSubdir);   // Managed binaries are in <Profiler-HOME>/net-ver-moniker/
-                if (binaryDirs != null && !String.IsNullOrWhiteSpace(managedBinariesDirectory))
-                {
-                    binaryDirs.Add(managedBinariesDirectory);
-                }
+                binaryDirs.Add(managedBinariesDirectory);
             }
 
-            {
-                // OPTION B from above (Profiler only):
-
-                string profilerHomeDirectory = nativeProfilerLoaderAssemblyDirectory;                           // Profiler-HOME
-                string managedBinariesDirectory = Path.Combine(profilerHomeDirectory, managedBinariesSubdir);   // Managed binaries are in <Profiler-HOME>/net-ver-moniker/
-
-                if (binaryDirs != null && !String.IsNullOrWhiteSpace(managedBinariesDirectory))
-                {
-                    binaryDirs.Add(managedBinariesDirectory);
-                }
-            }
+#if DEBUG
```

Comment:
Maybe use System.Diagnostics.Debugger.IsAttached so it would be possible to debug a RELEASE build

Reply:
✔️